xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/utilities/BaseSelect.vue

```diff
@@ -54,17 +54,18 @@
 
       <li
         v-for="option in options"
-        :key="option"
+        :key="option.value"
       >
         <a href="#"
           @click.prevent.stop="toggleSelect(option)">
           <slot
             name="option"
-            :value="option"
+            :value="option.value"
+            :label="option.name"
             :checked="isSelected(option)"
             :toggle="toggleSelect"
           >
-            {{ option }}
+            {{ option.name }}
           </slot>
         </a>
       </li>
```

Comment:
You also need to add the selected values that are not in the list of known/supported options. It's not common but it can happen that the filter / constraint value (either from the configuration given by the user as input or from the live data configuration saved in the URL) becomes obsolete, meaning that it's no longer a value that is accepted on new live data entries for that property, but that doesn't mean there can't be existing entries that are still using it and we shouldn't loose this configuration, not until the user selects a different value for the filter.

Reply:
Oh, I didn't thought about this use case!

But in this case, even if we still have the option value from the saved configuration, how do we get the option pretty name to display, as it is no longer in the LD meta config ?

Also, what happens when the user deselect the option from the widget? Is it supposed to disappear from the list of available options, or does it stays in the list, even if it is no longer an option and is not selected anymore?